com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ProtoBuilders.java

```diff
@@ -76,6 +81,58 @@ public static String replaceTextureSetName(String str) {
         return out;
     }
 
+    private static void validateMaterialAtlasCompatability(Project project, IResource resource, String material, String textureSet) throws IOException, CompileExceptionError {
```

Comment:
This function validates the compatibility of an atlas + material combination by looking at if the material shaders has any array samplers that can be used if the texture set is paged (and vice versa) which is done by parsing the shader code for the correct sampler keywords. Since this is done for every component that supports this feature this might be really slow.. Is there some way we can improve this process? Cache the validation result or something like that?